open_academy/models/session.py

```diff
@@ -1,14 +1,48 @@
-from odoo import models, fields
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
 
 
 class Session(models.Model):
     _name = "session"
     _description = "Session"
 
     name = fields.Char(required=True)
-    start_date = fields.Date()
+    start_date = fields.Date(default=lambda self: fields.Date.today(self))
     duration = fields.Float()
     seats = fields.Integer(string="Number of seats")
+    taken_seats = fields.Integer()
+    percent_seats = fields.Integer(string="Taken seats %", compute="_compute_seats")
+    active = fields.Boolean(default=True)
     instructor_id = fields.Many2one("res.partner", domain="[('instructor', '=', True)]")
     course_id = fields.Many2one("course")
     attendee_ids = fields.Many2many("res.partner")
+
+    @api.depends("seats", "taken_seats")
+    def _compute_seats(self):
+        for record in self:
+            if record.seats > 0:
+                record.percent_seats = round(record.taken_seats / record.seats * 100)
+            else:
+                record.percent_seats = 0
+
+    @api.onchange("seats", "taken_seats")
+    def _onchange_percent_seats(self):
+        if self.taken_seats > self.seats or self.taken_seats < 0 or self.seats < 0:
+            self.seats = self._origin.seats
+            self.taken_seats = self._origin.taken_seats
+
+            return {
+                "warning": {
+                    "title": "Invalid Values",
+                    "message": """Cannot have a negative number of seats or more seats taken than the total number
+of seats.""",
```

Comment:
No use `"""` for string messages, this could include unnecessary rendering spaces, also, use translation method `_`

```suggestion
                    "message": _("Cannot have a negative number of seats or more seats taken than the total number"
"of seats."),
```

Same for title